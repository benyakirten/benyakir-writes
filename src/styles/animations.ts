import { keyframes } from "styled-components";

export const rotate = keyframes`
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(-360deg);
    }
`

export const fadeIn = keyframes`
    from {
        opacity: 0%;
    }
    to {
        opacity: 100%;
    }
`
export const slideInLeft = keyframes`
    0% {
        transform: translateX(-40vw);
        opacity: 0.2;
    }
    40% {
        transform: translateX(4vw);
        opacity: 1;
    }
    100% {
        transform: translateX(0);
        opacity: 1;
    }
`

export const slideInRight = keyframes`
    0% {
        transform: translateX(40vw);
        opacity: 0.2;
    }
    80% {
        transform: translateX(-4vw);
        opacity: 1;
    }
    100% {
        transform: translateX(0);
        opacity: 1;
    }
`