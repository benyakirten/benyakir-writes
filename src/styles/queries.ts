export const mediaQuery = (maxWidth: number) => `(max-width: ${maxWidth}px)`;

export const DESKTOP_QUERY = mediaQuery(922);
export const TABLET_QUERY = mediaQuery(768);
export const PHONE_QUERY = mediaQuery(576);
export const DARK_MODE_QUERY = "(prefers-color-scheme: dark)";

export const media = {
  desktop: `@media ${DESKTOP_QUERY}`,
  tablet: `@media ${TABLET_QUERY}`,
  phone: `@media ${PHONE_QUERY}`,
  reducedMotion: "@media (prefers-reduced-motion)",
  noHover: "@media (any-hover: none)",
};
