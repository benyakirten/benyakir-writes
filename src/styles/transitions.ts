export const TIMEOUT_500 = 500;

export function getPageTransitionStyles(timeout: number): PageTransitionProperties {
  return {
    entering: {
      position: `absolute`,
      opacity: 0,
      tranform: 'translateY(-4rem)'
    },
    entered: {
      transition: `all ${timeout}ms ease-in-out`,
      opacity: 1,
      transform: 'translateY(0)'
    },
    exiting: {
      transition: `all ${timeout}ms ease-in-out`,
      opacity: 0,
      transform: 'translateY(4rem)'
    },
    exited: null,
    unmounted: null
  }
}