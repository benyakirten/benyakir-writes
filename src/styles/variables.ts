// FONT FAMILY
export const SERIF_FONT = "'Faustina', serif";
export const SANS_SERIF_FONT = "'Mulish', sans-serif";

// FONT SIZE
export const FONT_SIZE_XXXL = "2.25rem";
export const FONT_SIZE_XXL = "1.875rem";
export const FONT_SIZE_XL = "1.5rem";
export const FONT_SIZE_LG = "1.25rem";
export const FONT_SIZE_MD = "1.125rem";
export const FONT_SIZE_SM = "1rem";
export const FONT_SIZE_XS = "0.875rem";
export const FONT_SIZE_XXS = "0.75rem";

// LINE HEIGHT
export const LINE_HEIGHT_XXXL = "2.5rem";
export const LINE_HEIGHT_XXL = "2.25rem";
export const LINE_HEIGHT_XL = "2rem";
export const LINE_HEIGHT_LG = "1.75rem";
export const LINE_HEIGHT_MD = "1.75rem";
export const LINE_HEIGHT_SM = "1.5rem";
export const LINE_HEIGHT_XS = "1.25rem";
export const LINE_HEIGHT_XXS = "1rem";

// FONT
export const FONT_XXXL = `font-size: ${FONT_SIZE_XXXL}; line-height: ${LINE_HEIGHT_XXXL};`;
export const FONT_XXL = `font-size: ${FONT_SIZE_XXL}; line-height: ${LINE_HEIGHT_XXL};`;
export const FONT_XL = `font-size: ${FONT_SIZE_XL}; line-height: ${LINE_HEIGHT_XL};`;
export const FONT_LG = `font-size: ${FONT_SIZE_LG}; line-height: ${LINE_HEIGHT_LG};`;
export const FONT_MD = `font-size: ${FONT_SIZE_MD}; line-height: ${LINE_HEIGHT_MD};`;
export const FONT_SM = `font-size: ${FONT_SIZE_SM}; line-height: ${LINE_HEIGHT_SM};`;
export const FONT_XS = `font-size: ${FONT_SIZE_XS}; line-height: ${LINE_HEIGHT_XS};`;
export const FONT_XXS = `font-size: ${FONT_SIZE_XXS}; line-height: ${LINE_HEIGHT_XXS};`;

// COLOR
export const PRIMARY_100 = "#FDF3D8";
export const PRIMARY_200 = "#FDEDC4";
export const PRIMARY_300 = "#FDE6B0";
export const PRIMARY_400 = "#FBD989";
export const PRIMARY_500 = "#F9C74F";
export const PRIMARY_600 = "#FBD989";
export const PRIMARY_700 = "#F7B926";
export const PRIMARY_800 = "#FF7000";
export const PRIMARY_900 = "#A34700";

export const SECONDARY_100 = "#D8E2FD";
export const SECONDARY_200 = "#9EB7FA";
export const SECONDARY_300 = "#90A6E3";
export const SECONDARY_400 = "#8196CD";
export const SECONDARY_500 = "#7385B6";
export const SECONDARY_600 = "#65749f";
export const SECONDARY_700 = "#566488";
export const SECONDARY_800 = "#485372";
export const SECONDARY_900 = "#39435b";

export const GRAY_100 = "#f8f9fa";
export const GRAY_200 = "#e9ecef";
export const GRAY_300 = "#dee2e6";
export const GRAY_400 = "#ced4da";
export const GRAY_500 = "#adb5bd";
export const GRAY_600 = "#6c757d";
export const GRAY_700 = "#495057";
export const GRAY_800 = "#343a40";
export const GRAY_900 = "#212529";

export const BLACK = "#05491F";
export const WHITE = "#D8E3FD";
export const ABS_BLACK = "#000";
export const ABS_WHITE = "#FFF";

export const SUCCESS = "#44764d";
export const ERROR = "#993d4d";

// SIZE
export const SIZE_XXL = "4rem";
export const SIZE_XL = "3rem";
export const SIZE_LG = "2.4rem";
export const SIZE_MD = "1.6rem";
export const SIZE_SM = "0.8rem";
export const SIZE_XS = "0.4rem";

// PADDING/MARGIN
export const HORIZONTAL_XL = "0 3rem";
export const HORIZONTAL_LG = "0 2.4rem";
export const HORIZONTAL_MD = "0 1.6rem";
export const HORIZONTAL_SM = "0 0.8rem";
export const HORIZONTAL_XS = "0 0.4rem";

export const VERTICAL_XL = "3rem 0";
export const VERTICAL_LG = "2.4rem 0";
export const VERTICAL_MD = "1.6rem 0";
export const VERTICAL_SM = "0.8rem 0";

export const BOTH_XL = "2.4rem 3rem";
export const BOTH_LG = "2rem 2.4rem";
export const BOTH_MD = "1.4rem 1.6rem";
export const BOTH_SM = "0.6rem 1rem";

// BORDER SHADOW
export const SHADOW_XL = "0 1.6rem 2.4rem";
export const SHADOW_LG = "0 0.3rem 0.6rem";
export const SHADOW_MD = "3px 2px 4px";
export const SHADOW_MD_BALANCED = "0px 2px 4px";
export const SHADOW_SM = "0 1px 2px";
export const SHADOW_UP = "-2px -3px 1.2rem -2px rgba(0,0,0,0.75)";
export const SHADOW_CARD_NORMAL = "2px 6px 0px 0px rgba(0,0,0,0.75);";
export const SHADOW_CARD_HOVER = "2px 16px 0px 0px rgba(0,0,0,0.75);";

// Z INDEX
export const Z_SEARCH = 6;
export const Z_SEARCH_BACKDROP = 6;
export const Z_TOP = 4;
export const Z_HIGH = 3;
export const Z_RAISED = 2;
export const Z_ABOVE = 1;
export const Z_BASE = 0;
export const Z_UNDER = -1;

// TRANSITION TIMING
export const TRANSITION_FAST = "100ms";
export const TRANSITION_NORMAL = "200ms";
export const TRANSITION_SLOW = "300ms";
export const TRANSITION_EXTRA_SLOW = "500ms";
