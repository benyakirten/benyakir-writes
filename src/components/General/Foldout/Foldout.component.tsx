import * as React from "react"

import { FoldoutContainer, FoldoutBody } from "./Foldout.styles"

import DownArrow from "@Pre/DownArrow/DownArrow.component"

import { FoldoutProps } from "@Types/props"

const Foldout: React.FC<FoldoutProps> = ({
    open = false,
    height = "4rem",
    topbar,
    onClick,
    children,
    heightMultiplierOnPhone,
    heightMultiplierOnTablet,
    heightMultiplierOnLarger,
    cyId
}) => {
    const handleContainerClick = React.useCallback((e: React.BaseSyntheticEvent) => _handleContainerClick(e), [onClick])
    function _handleContainerClick(e: React.BaseSyntheticEvent) {
        if (
            e.target.getAttribute("data-navtoggle") === "no-toggle" ||
            (e.target.parentElement && e.target.parentElement.getAttribute("data-navtoggle") === "no-toggle") ||
            (e.target.parentElement.parentElement && e.target.parentElement.parentElement.getAttribute("data-navtoggle") === "no-toggle") ||
            (e.target.parentElement.parentElement.parentElement && e.target.parentElement.parentElement.parentElement.getAttribute("data-navtoggle") === 'no-toggle')
        ) {
            return
        }
        onClick && onClick()
    }
    return (
        <FoldoutContainer open={open} onClick={(e) => handleContainerClick(e)}>
            <DownArrow open={open} tabIndex={open ? 0 : -1} onClick={onClick} cyId={cyId ? cyId : "foldout-bar"} />
            {topbar}
            <FoldoutBody
                data-navtoggle="no-toggle"
                heightMultiplierOnPhone={heightMultiplierOnPhone}
                heightMultiplierOnTablet={heightMultiplierOnTablet}
                heightMultiplierOnLarger={heightMultiplierOnLarger}
                height={height}
                open={open}
                aria-hidden={!open}
            >
                {children}
            </FoldoutBody>
        </FoldoutContainer>
    )
}

export default Foldout
