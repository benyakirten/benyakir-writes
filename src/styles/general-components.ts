import styled from "styled-components";

import { fadeIn, slideInLeft, slideInRight } from "./animations";
import { media } from "./queries";
import {
	FONT_SIZE_MD,
	SIZE_MD,
	TRANSITION_EXTRA_SLOW,
	TRANSITION_NORMAL,
	SANS_SERIF_FONT,
	SIZE_SM,
	SIZE_XS,
	SIZE_LG,
	FONT_SIZE_XXS,
	FONT_XXXL,
	FONT_XXL,
	FONT_XL,
	FONT_MD,
	FONT_LG,
} from "./variables";

import { hoverUnderline, textClipWithHoverTransition } from "./style-mixins";

export const FadeIn = styled.div<{ duration?: string; delay?: string }>`
  opacity: 0;
  animation: ${fadeIn}
    ${(props) => (props.duration ? props.duration : TRANSITION_NORMAL)}
    ${(props) => (props.delay ? props.delay : "0ms")} ease forwards;
`;

export const SlideInLeft = styled.div<{ duration?: string; delay?: string }>`
  opacity: 0;
  transform: translateX(-40vw);
  animation: ${slideInLeft}
    ${(props) => (props.duration ? props.duration : TRANSITION_EXTRA_SLOW)}
    ${(props) => (props.delay ? props.delay : "0ms")} ease forwards;
`;

export const SlideInRight = styled.div<{ duration?: string; delay?: string }>`
  opacity: 0;
  transform: translateX(40vw);
  animation: ${slideInRight}
    ${(props) => (props.duration ? props.duration : TRANSITION_EXTRA_SLOW)}
    ${(props) => (props.delay ? props.delay : "0ms")} ease forwards;
`;

export const Grouping = styled.section<{ marginVertical?: string }>`
  position: relative;
  margin-top: ${(props) => (props.marginVertical ? props.marginVertical : "0")};
  margin-bottom: ${(props) =>
		props.marginVertical ? props.marginVertical : "4rem"};
`;

// TODO: Make the colors look good
export const LeadHeading = styled.h1`
  position: relative;

  margin-bottom: ${SIZE_MD};

  ${FONT_XXXL};

  ${(props) => textClipWithHoverTransition(props.theme.header.startColor, props.theme.header.endColor)}

  ${media.phone} {
    ${FONT_XXL};
  }
`;

export const Subtitle = styled.h2`
  position: relative;

  margin-bottom: ${SIZE_MD};

  ${FONT_XL};

  ${(props) => textClipWithHoverTransition(props.theme.header.startColor, props.theme.header.endColor)}

  ${media.phone} {
    ${FONT_LG};
  }
`;

export const BigParagraph = styled.p`
  ${FONT_XL};
`;

export const WpContentDescription = styled.div<{ fontSize?: string }>`
  ${FONT_MD};

  overflow: hidden;
  text-overflow: ellipsis;

  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;

  a:link,
  a:visited {
    position: relative;

    color: ${(props) => props.theme.base.textColor};

    overflow: hidden;
    white-space: nowrap;

    &::after {
      content: '';

      position: absolute;
      left: 0;
      bottom: 0;

      height: 2px;
      width: 100%;

      background-color: ${(props) => props.theme.base.background};

      transition: transform ${TRANSITION_NORMAL} ease;
      transform-origin: left;
      transform: scaleX(0);
    }

    &:hover {
      &::after {
        transform: scaleX(1);

        ${media.reducedMotion} {
          transform: scaleX(0);
        }
      }
    }
  }
`;

export const Card = styled.div`
  border-radius: ${SIZE_XS};
  border: 1px solid ${(props) => props.theme.base.border};

  padding: ${SIZE_SM};
  background-color: ${(props) => props.theme.base.background};
`;

export const Box = styled.div`
  display: flex;
  align-items: center;
  gap: ${SIZE_MD};
`;

export const FlatBox = styled(Box)`
  margin: 0;
`;

// https://stackoverflow.com/questions/24933430/img-src-svg-changing-the-styles-with-css
export const SVGContainer = styled.div<{
	src: string;
	alt: string;
	size?: string;
}>`
  position: relative;

  background-color: ${(props) => props.theme.base.textColor};
  mask: url(${(props) => props.src}) no-repeat center;

  height: ${(props) => props.size ?? SIZE_MD};
  width: ${(props) => props.size ?? SIZE_MD};
`;

export const WpContent = styled.div`
  ${FONT_LG};

  ${media.phone} {
    ${FONT_MD};
  }

  a:link,
  a:visited {
    ${(props) => hoverUnderline(props.theme.base.link)}
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${SANS_SERIF_FONT};
    padding: ${SIZE_MD} 0;
    ${(props) => textClipWithHoverTransition(props.theme.header.startColor, props.theme.header.endColor)}
  }


  ul,
  ol,
  li {
    margin-left: ${SIZE_SM};
  }

  ul:has(li > a), ol:has(li > a) {
    list-style-type: disc;
  }

  p {
    margin: ${SIZE_MD} 0;
  }

  code {
    font-family: monospace;
    font-size: ${FONT_SIZE_MD};
  }
`;

export const Page = styled.div`
  overflow-y: auto;
  height: 100vh;
`;

export const PageContents = styled.div`
  padding: 2rem 4rem;
  ${media.tablet} {
    padding: 1rem 4rem;
  }
`;

export const PillContainer = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;

  white-space: nowrap;
	
  height: min-content;
  width: min-content;

	font-family: ${SANS_SERIF_FONT};
  font-size: ${FONT_SIZE_XXS};
  text-transform: capitalize;
  
  border-radius: ${SIZE_LG};
  color: ${(props) => props.theme.base.textColor};
  background-color: ${(props) => props.theme.pill.background};

	padding: ${SIZE_XS} ${SIZE_SM};

  ${media.phone} {
    padding: ${SIZE_XS};
  }
`;

export const PaginatedPageContents = styled.div`
  padding: 2rem 18rem 4rem 1rem;
  ${media.phone} {
    padding: 2rem 0.5rem 4rem 0.5rem;
  }
`;

export const NormalPageContents = styled.div`
  padding: 2rem 1rem;
  ${media.phone} {
    padding: 2rem 0.5rem;
  }
`;

export const TemplateHeaderTitle = styled(LeadHeading)`
  margin-bottom: 0;
`;

export const TemplateContent = styled.div`
  margin: ${SIZE_MD} 0;
`;

export const TemplateHeaderContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${SIZE_XS};
  margin-bottom: ${SIZE_SM};
`;
