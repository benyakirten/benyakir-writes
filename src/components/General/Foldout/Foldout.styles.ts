import styled from "styled-components";

import { TRANSITION_NORMAL } from "@/styles/variables";
import { multiplyCSSNumber } from "@/utils/strings";
import { media } from "@/styles/queries";

export const FoldoutContainer = styled.div`
  position: relative;
`;

export const TopbarContainer = styled.div`
  cursor: pointer;
  max-width: max-content;
`;

export const FoldoutBody = styled.div<{
	open: boolean;
	height: string;
	heightMultiplierOnPhone?: number;
	heightMultiplierOnTablet?: number;
	heightMultiplierOnLarger?: number;
}>`
  display: flex;
  flex-direction: column;

  cursor: default;

  height: ${(props) => (props.open ? props.height : "0")};

  transform-origin: top;
  transform: scaleY(${(props) => (props.open ? "1" : "0.8")});
  transition: all ${TRANSITION_NORMAL} ease;

  opacity: ${(props) => (props.open ? "1" : "0")};

  ${media.custom(1160)} {
    height: ${(props) =>
			props.open
				? multiplyCSSNumber(
						props.height,
						props.heightMultiplierOnLarger ? props.heightMultiplierOnLarger : 1,
					)
				: "0"};
  }

  ${media.desktop} {
    height: ${(props) =>
			props.open
				? multiplyCSSNumber(
						props.height,
						props.heightMultiplierOnTablet ? props.heightMultiplierOnTablet : 1,
					)
				: "0"};
  }
  ${media.phone} {
    height: ${(props) =>
			props.open
				? multiplyCSSNumber(
						props.height,
						props.heightMultiplierOnPhone ? props.heightMultiplierOnPhone : 1.2,
					)
				: "0"};
  }

  ${media.reducedMotion} {
    transition: all 1ms ease !important;
  }
`;
