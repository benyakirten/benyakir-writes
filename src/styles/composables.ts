import { media } from "./queries";
import { TRANSITION_EXTRA_SLOW, TRANSITION_SLOW } from "./variables";

export function textClipWithHoverTransition(
	startColor: string,
	endColor: string,
) {
	return `
        background-image: linear-gradient(45deg, ${startColor} 45%, ${endColor});
        background-position: left;
        background-size: 300%;
        background-clip: text;
        color: transparent;
        transition: background-position ${TRANSITION_EXTRA_SLOW} ease;

        &:hover {
            background-position: right;
        }
    `;
}

export function hoverUnderline(color: string) {
	return `
        color: ${color};

        width: min-content;

        background-color: transparent;
        background-image: linear-gradient(${color}, ${color});
        background-repeat: no-repeat;
        background-size: 0px 2px;
        background-position: 0 100%;
        text-decoration: none;

        transition: background-size ${TRANSITION_SLOW} ease;

        &:hover, &:focus-within {
            background-size: 100% 2px;
        }

        &:focus-within {
            outline: 1px solid ${color};
        }

        ${media.noHover} {
            background-size: 100% 2px;
        }
    `;
}
