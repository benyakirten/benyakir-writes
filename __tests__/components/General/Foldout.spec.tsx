import * as React from "react";
import { cleanup, render, screen, fireEvent } from "@testing-library/react";

import renderer from "react-test-renderer";
import "jest-styled-components";

import Foldout from "@Gen/Foldout/Foldout.component";
import { FoldoutBody } from "@Gen/Foldout/Foldout.styles";

describe("Foldout component", () => {
    const clickSpy = jest.fn();

    beforeEach(() => {
        clickSpy.mockClear();
    });

    afterEach(cleanup);

    it("should render properly", () => {
        expect(() => render(<Foldout topbar={<p>Hello</p>} />)).not.toThrow();
    });

    it("should set the down arrow's tabindex according to the open prop", async () => {
        render(<Foldout topbar={<p>Test Topbar</p>} open={false} />);
        const downArrowOne = await screen.getByRole("button");
        expect(downArrowOne.getAttribute("tabindex")).toEqual("-1");

        cleanup();

        render(<Foldout topbar={<p>Test Topbar</p>} open={true} />);
        const downArrowTwo = await screen.getByRole("button");
        expect(downArrowTwo.getAttribute("tabindex")).toEqual("0");
    });

    it("should adjust the aria-hidden attribute of the foldout body according to the open prop", async () => {
        let topbar, body;

        render(<Foldout topbar={<p>Test Topbar</p>} open={false} />);
        topbar = await screen.getByText("Test Topbar");
        body = topbar.nextElementSibling as HTMLElement;
        expect(body?.getAttribute("aria-hidden")).toEqual("true");

        cleanup();

        render(<Foldout topbar={<p>Test Topbar</p>} open={true} />);
        topbar = await screen.getByText("Test Topbar");
        body = topbar.nextElementSibling;
        expect(body?.getAttribute("aria-hidden")).toEqual("false");
    });

    it("should dispatch the onClick event if the foldout or any element within it is clicked unless the item or its parent has an id of no-toggle such as the body and its descendants", async () => {
        render(
            <Foldout
                topbar={<p>Test Topbar</p>}
                open={false}
                onClick={clickSpy}
            >
                <p>Some stuff</p>
            </Foldout>
        );
        const button = await screen.getByRole("button");
        fireEvent.click(button);
        expect(clickSpy).toHaveBeenCalledTimes(2);

        const topbar = await screen.getByText("Test Topbar");
        fireEvent.click(topbar);
        expect(clickSpy).toHaveBeenCalledTimes(3);

        const body = topbar.nextElementSibling!;
        expect(body.getAttribute('data-navtoggle')).toEqual("no-toggle");
        fireEvent.click(body);
        expect(clickSpy).toHaveBeenCalledTimes(3);

        const stuff = await screen.getByText("Some stuff");
        fireEvent.click(stuff);
        expect(clickSpy).toHaveBeenCalledTimes(3);
    });

    it("should render the outer div's cursor based on the open prop", () => {
        const outer = renderer.create(
            <Foldout topbar={<p>Test Topbar</p>} open={true} />
        );
        expect(outer.toJSON()).toMatchSnapshot();
        expect(outer.toJSON()).toHaveStyleRule("cursor", "n-resize");

        const innerOne = (outer.toJSON()! as any).children[2]
        expect(innerOne).toHaveStyleRule("height", "4rem")
        expect(innerOne).toHaveStyleRule("opacity", "1")


        outer.update(<Foldout topbar={<p>Test Topbar</p>} open={false} />);
        expect(outer.toJSON()).toHaveStyleRule("cursor", "s-resize");

        const innerTwo = (outer.toJSON()! as any).children[2]
        expect(innerTwo).toHaveStyleRule("height", "0")
        expect(innerTwo).toHaveStyleRule("opacity", "0")
    });
});
