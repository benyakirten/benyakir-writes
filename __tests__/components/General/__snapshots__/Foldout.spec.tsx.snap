// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foldout component should render the outer div's cursor based on the open prop 1`] = `
.c0 {
  position: relative;
  cursor: n-resize;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  height: 4rem;
  opacity: 1;
}

.c1 {
  position: absolute;
  left: -1.5rem;
  top: 0.4rem;
  outline: none;
  font-size: 2.4rem;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

@media (max-width:1160px) {
  .c2 {
    height: 4rem;
  }
}

@media (max-width:922px) {
  .c2 {
    height: 4rem;
  }
}

@media (max-width:576px) {
  .c2 {
    height: 4rem;
  }
}

@media (prefers-reduced-motion) {
  .c2 {
    -webkit-transition: all 0.01s ease !important;
    transition: all 0.01s ease !important;
  }
}

<div
  className="c0"
  onClick={[Function]}
  open={true}
>
  <span
    className="c1"
    open={true}
    role="button"
    tabIndex={0}
  >
    â†¡
  </span>
  <p>
    Test Topbar
  </p>
  <div
    aria-hidden={false}
    className="c2"
    data-navtoggle="no-toggle"
    height="4rem"
    open={true}
  />
</div>
`;
